components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3936,4 +3937,37 @@ public static String getInternalEveryoneRole(AuthenticatedUser user) throws Iden
             throw new IdentityOAuth2Exception(errorMsg, e);
         }
     }
+
+    /**
+     * Replace placeholder values of the url with server configurations
+     *
+     * @param url
+     * @return
+     */
+    public static String processURI(String url) {
+
+        if (StringUtils.contains(url, IdentityConstants.CarbonPlaceholders.CARBON_PORT)) {
+
+            String mgtTransport = CarbonUtils.getManagementTransport();
+            AxisConfiguration axisConfiguration = OAuth2ServiceComponentHolder.getInstance()
+                    .getConfigurationContextService().
+                            getServerConfigContext().getAxisConfiguration();
+            int mgtTransportPort = CarbonUtils.getTransportProxyPort(axisConfiguration, mgtTransport);
+            if (mgtTransportPort <= 0) {
+                mgtTransportPort = CarbonUtils.getTransportPort(axisConfiguration, mgtTransport);
+            }
+            // If it's well known HTTPS port, skip adding port
+            if (mgtTransportPort == IdentityCoreConstants.DEFAULT_HTTPS_PORT) {
+                url = StringUtils.replace(url, ":" +
+                        IdentityConstants.CarbonPlaceholders.CARBON_PORT, "");
+            }
+
+        }
```

Comment:
We can move this to a private method and use it within the if-block that checks for place holders